platform/npm/tests/request-data.js

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+
+    uBlock Origin - a browser extension to block requests.
+    Copyright (C) 2014-present Raymond Hill
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see {http://www.gnu.org/licenses/}.
+
+    Home: https://github.com/gorhill/uBlock
+*/
+
+'use strict';
+
+/******************************************************************************/
+
+import { strict as assert } from 'assert';
+import { readFile } from 'fs/promises';
+import { createRequire } from 'module';
+import { dirname, resolve } from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const require = createRequire(import.meta.url);
+
+const requests = require('scaling-palm-tree/requests.json');
+const results = require('../results.json');
+
+const typeMap = {
+    document: 'sub_frame',
+    stylesheet: 'stylesheet',
+    image: 'image',
+    media: 'media',
+    font: 'font',
+    script: 'script',
+    xhr: 'xmlhttprequest',
+    fetch: 'xmlhttprequest',
+    websocket: 'websocket',
+    ping: 'ping',
+
+    other: 'other',
+    eventsource: 'other',
+    manifest: 'other',
+    texttrack: 'other',
+};
+
+async function read(path) {
+    return readFile(resolve(__dirname, path), 'utf8');
+}
+
+describe('Request data', () => {
+    let engine = null;
+
+    before(async () => {
+        const globals = { URL, setTimeout, clearTimeout };
+
+        const { StaticNetFilteringEngine } = await import('../index.js');
+
+        engine = await StaticNetFilteringEngine.create();
+
+        await engine.useLists([
```

Comment:
@gorhill based on this comment: https://github.com/gorhill/uBlock/pull/3828#issuecomment-899120180